src/libraries/Common/src/Interop/Unix/System.Native/Interop.IPAddress.cs

```diff
@@ -73,16 +73,9 @@ public bool Equals(IPAddress other)
 
                 fixed (byte* thisAddress = Address)
                 {
-                    for (int i = 0; i < addressByteCount; i++)
-                    {
-                        if (thisAddress[i] != other.Address[i])
-                        {
-                            return false;
-                        }
-                    }
+                    return new ReadOnlySpan<byte>(thisAddress, addressByteCount).SequenceEqual(
```

Comment:
Would using MemoryMarshal.CreateSpan here to avoid pinning make sense?

Reply:
How would it avoid the need for fixed?

Reply:
You could do `MemoryMarshal.CreateReadOnlySpan(ref Address[0], addressByteCount)`.

Reply:
If you do that, `fixed` in GetHashCode should get the same treatment.

Reply:
> You could do `MemoryMarshal.CreateReadOnlySpan(ref Address[0], addressByteCount)`.

Sure, it could.

Reply:
I'll look at changing these separately.

Reply:
https://github.com/dotnet/runtime/pull/71317